### PR TITLE
Sean Valdivia. Still needing help on understanding closure.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/arrays.js [hunk 1]
```diff
@@ -9,35 +9,60 @@ const each = (elements, cb) => {
...
+  return newArr;
 };
 
 const reduce = (elements, cb, startingValue) => {
   // Combine all elements into a single value going from left to right.
   // Elements will be passed one by one into `cb`.
   // `startingValue` is the starting value.  If `startingValue` is undefined then make `elements[0]` the initial value.
+  if (typeof startingValue === 'undefined') {
+    startingValue = elements[0];
+  }
```
Comment (sperrye) [inline src/arrays.js:33]: The next steps would be to somehow loop over the elements, pass them to cb, and add them to startingValue.

File: src/callbacks.js [hunk 2]
```diff
@@ -22,36 +22,55 @@
...
 
 sumNums(5, 10, (sum) => {
   console.log(`The sum is ${sum}.`);
 });
 
 // Write a function called multiplyNums that adds two numbers and passes the result to the callback
 // code here
+const multiplyNums = (x, y, cb) => {
+  const product = x * y;
+  cb(product);
```
Comment (sperrye) [inline src/callbacks.js:72]: You can also pass the operation to cb without initializing a new variable `cb(x*y)`

File: src/callbacks.js [hunk 3]
```diff
@@ -68,13 +96,25 @@ contains(foods, 'ribeye', (result) => {
 // Write a function called removeDuplicates that removes all duplicate values from the given array.
 // Pass the array to the callback function.  Do not mutate the original array.
 // code here
+const removeDuplicates = (arr, cb) => {
+  var newArr = arr.filter(function(el, i, ar) {
+    return ar.indexOf(el) === i;
+  });
```
Comment (sperrye) [inline src/callbacks.js:102]: Neat!

File: src/closure.js [hunk 4]
```diff
@@ -5,17 +5,44 @@ const counter = () => {
...
 };
 
 const counterFactory = () => {
   // Return an object that has two methods called `increment` and `decrement`.
   // `increment` should increment a counter variable in closure scope and return it.
   // `decrement` should decrement the counter variable and return it.
+  let count = 0;
+  const counterObj = {};
+  counterObj.increment = () => {
+    count += 1;
```
Comment (sperrye) [inline src/closure.js:23]: You can also use the increment/decrement operators `++/--`

**Review by sperrye — COMMENTED** [review]
Complete, great job

File: src/callbacks.js (hunk 3 above)
Comment (ghost) [reply src/callbacks.js:102]: I can't believe I forgot the other r on the ar.indexOf.

**Pull request closed without merging**